votor/src/consensus_metrics.rs

```diff
@@ -97,54 +118,103 @@ pub struct ConsensusMetrics {
     node_metrics: BTreeMap<Pubkey, NodeVoteMetrics>,
     /// Used to track when this node received blocks from different leaders in the network.
     leader_metrics: BTreeMap<Pubkey, Histogram>,
+    /// Counts number of times metrics recording failed.
+    metrics_recording_failed: usize,
     /// Tracks when individual slots began.
     ///
     /// Relies on [`TimerManager`] to notify of start of slots.
     /// The manager uses parent ready event and timeouts as per the Alpenglow protocol to determine start of slots.
     start_of_slot: BTreeMap<Slot, Instant>,
     /// Tracks the current epoch, used for end of epoch reporting.
     current_epoch: Epoch,
+    /// Receiver for events
+    receiver: ConsensusMetricsEventReceiver,
 }
 
 impl ConsensusMetrics {
-    pub fn new(epoch: Epoch) -> Self {
+    fn new(epoch: Epoch, receiver: ConsensusMetricsEventReceiver) -> Self {
         Self {
             node_metrics: BTreeMap::default(),
             leader_metrics: BTreeMap::default(),
+            metrics_recording_failed: 0,
             start_of_slot: BTreeMap::default(),
             current_epoch: epoch,
+            receiver,
+        }
+    }
+
+    pub fn start_metrics_loop(
+        epoch: Epoch,
+        receiver: ConsensusMetricsEventReceiver,
+        exit: Arc<AtomicBool>,
+    ) {
+        Builder::new()
+            .name("consensus-metrics".into())
+            .spawn(move || {
+                let mut metrics = Self::new(epoch, receiver);
```

Comment:
Similarly, if we keep the thread, can you add `info!` logs to indicate start and stop. Ie,
```rust
info!("ConsensusMetricsService has started");
let mut metrics = ...;
metrics.run();
info!("ConsensusMetricsService has stopped");
```

Reply:
Done.